java6/src/main/java/io/papermc/paperclip/Main.java

```diff
@@ -14,7 +14,8 @@
 public final class Main {
 
     public static void main(final String[] args) {
-        if (getJavaVersion() < 17) {
+        float javaVersion = Float.parseFloat(System.getProperty("java.class.version"));
```

Comment:
The version is not a float. It is two components of a 16-bit unsigned integer for minor and the same for major. This should reflect that properly.